### PR TITLE
I created  and implemented delete account end point and flow and wrot…

### PR DESCRIPTION
**Pull request by M-Jhay**
…e unit test  to test my code if it works as expected

### CODE REVIEW

**Review by giftedGeorge — DISMISSED** [review]
looking good.

File: src/DwitTech.AccountService.Data/repository/UserRepository.cs [hunk 1]
```diff
@@ -81,6 +81,13 @@ public async Task UpdateUserLoginAsync(User user, string newPasswordHash)
                 await _dbContext.SaveChangesAsync();
             }
         }
+
+        public async Task DeleteUserAsync(int id)
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Data/repository/UserRepository.cs:85]: Please implement a soft delete.

File: tests/DwitTech.AccountService.Core.Tests/Services/UserServiceTest.cs [hunk 2]
```diff
@@ -285,5 +286,60 @@ public async Task Logout_Should_Return_True_If_User_Was_Successfully_Logged_Out(
...
+
+            // Act
+            await userService.DeleteUserAsync(userIdToDelete);
+
+            // Assert
+            iUserRepoMock.Verify(repo => repo.DeleteUserAsync(userIdToDelete), Times.Once);
+        }
+
+        [Fact]
+        public async Task DeleteUserAsync_ThrowsDbUpdateException_WhenDatabaseUpdateFails()
```
Comment (giftedGeorge) [inline tests/DwitTech.AccountService.Core.Tests/Services/UserServiceTest.cs:320]: I think this test should be in your repo test, not here.

File: tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs [hunk 3]
```diff
@@ -353,5 +353,81 @@ public async Task Logout_Should_Return_False_If_AuthorizationHeader_Is_Not_Given
             Assert.IsType<BadRequestObjectResult>(result);
 
         }
+
+        [Fact]
+        public async Task DeleteUser_Returns_NoContentResult()
```
Comment (giftedGeorge) [inline tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs]: I think you forgot to add "WhenSuccessful" in your method name here.

File: tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs [hunk 4]
```diff
@@ -353,5 +353,81 @@ public async Task Logout_Should_Return_False_If_AuthorizationHeader_Is_Not_Given
...
+            var iloggerMock = new Mock<ILogger<UserController>>();
+            var _mockAuthService = new Mock<IAuthenticationService>();
+            var _mockUserService = new Mock<IUserService>();
+
+            var userController = new UserController(_activationService.Object, _mockAuthService.Object, _mockUserService.Object, iloggerMock.Object);
+
+            // Act
+            var result = await userController.DeleteUser(userId);
+
+            // Assert
```
Comment (giftedGeorge) [inline tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs:375]: How is this test successful without you setting up the delete method in the service?

File: tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs [hunk 5]
```diff
@@ -353,5 +353,81 @@ public async Task Logout_Should_Return_False_If_AuthorizationHeader_Is_Not_Given
...
+                    (Func<It.IsAnyType, Exception, string>)It.IsAny<object>()),
+                Times.Once);
+        }
+
+        [Fact]
+        public async Task DeleteUser_WithInvalidId_ReturnsNotFound()
+        {
+            // Arrange
+            var id = 1;
+            var exceptionMessage = $"User with ID {id} not found";
```
Comment (giftedGeorge) [inline tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs:393]: This variable should be modified as needed and used in the assert.Equal statement below.

File: tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs [hunk 6]
```diff
@@ -353,5 +353,81 @@ public async Task Logout_Should_Return_False_If_AuthorizationHeader_Is_Not_Given
...
+            var notFoundResult = Assert.IsType<NotFoundObjectResult>(result);
+            Assert.Equal("User with ID 1 not found", notFoundResult.Value);
+        }
+
+        [Fact]
+        public async Task DeleteUser_WithException_ReturnsStatusCode500()
+        {
+            // Arrange
+            var id = 1;
+            var errorMessage = "Something went wrong";
```
Comment (giftedGeorge) [inline tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs:417]: This variable was not used in the test. Perhaps there should be an assert statement to verify that the message in the variable matches the message generated in the act.

**Review by giftedGeorge — DISMISSED** [review]
Please check the comments and make changes where necessary.

File: tests/DwitTech.AccountService.Core.Tests/Services/UserServiceTest.cs (hunk 2 above)
Comment (M-Jhay) [reply tests/DwitTech.AccountService.Core.Tests/Services/UserServiceTest.cs:320]: > I think this test should be in your repo test, not here.

Did you check my DeleteUserAsync implementation in my user service? I did a catch for DbUpdateException, hence the test.

File: tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs (hunk 3 above)
Comment (M-Jhay) [reply tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs]: > I think you forgot to add "WhenSuccessful" in your method name here.

Resolved

File: tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs (hunk 4 above)
Comment (M-Jhay) [reply tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs:375]: > How is this test successful without you setting up the delete method in the service?

Resolved

File: tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs (hunk 5 above)
Comment (M-Jhay) [reply tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs:393]: > This variable should be modified as needed and used in the assert.Equal statement below.

Resolved

File: tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs (hunk 6 above)
Comment (M-Jhay) [reply tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs:417]: > This variable was not used in the test. Perhaps there should be an assert statement to verify that the message in the variable matches the message generated in the act.

Resolved

**Comment by M-Jhay** [comment]
> Please check the comments and make changes where necessary.

Thank you for your observation. Kindly note that all comments have been resolved. I would appreciate it if you review my PR once more and approve it. Thank you

**Review by johnny2565 — APPROVED** [review]
The PR looks okay to me

**Review by dabensymon — APPROVED** [review]

**Pull request merged**